relay-server/src/endpoints/common.rs

```diff
@@ -530,15 +532,29 @@ where
         .or_else(move |error: BadStoreRequest| {
             metric!(counter(RelayCounters::EnvelopeRejected) += 1);
 
-            if let Some(category) = *event_category.borrow() {
-                if let Some(outcome) = error.to_outcome() {
+            let envelope_summary = envelope_summary.borrow();
+            if let Some(outcome) = error.to_outcome() {
+                if let Some(category) = envelope_summary.event_category {
                     outcome_producer.do_send(TrackOutcome {
                         timestamp: start_time,
                         scoping: *scoping.borrow(),
-                        outcome,
+                        outcome: outcome.clone(),
                         event_id: *event_id.borrow(),
                         remote_addr,
                         category,
+                        quantity: 1,
+                    });
+                }
+
+                if envelope_summary.attachment_quantity > 0 {
+                    outcome_producer.do_send(TrackOutcome {
+                        timestamp: start_time,
+                        scoping: *scoping.borrow(),
+                        outcome,
```

Comment:
why doesn't this outcome need a `.clone`?

Reply:
This is the final use of the `outcome` variable, so it can be moved into `TrackOutcome`. There is no more access to it afterwards. If there was, then you'd have to `.clone()` here, too.